client/my-sites/checkout/composite-checkout/components/wp-checkout-order-summary.js

```diff
@@ -146,6 +159,12 @@ function CheckoutSummaryFeaturesList( props ) {
 		refundDays = 4;
 	} else if ( hasPlanInCart && ! hasDomainsInCart ) {
 		refundDays = hasMonthlyPlan ? 7 : 14;
+	} else if ( isPaidSite && sitePlan ) {
+		if ( [ TERM_ANNUALLY, TERM_BIENNIALLY ].some( ( t ) => t === sitePlan.term ) ) {
+			refundDays = 14;
+		} else if ( sitePlan.term === TERM_MONTHLY ) {
+			refundDays = 7;
+		}
 	}
```

Comment:
I think we have a logic issue here. We're describing the refund window _for the items in the cart_. As such, we shouldn't be describing the refund window based on the existing site plan.

We have also determined via the `if` and `else if` above that we do _not_ have the following situations:
1. We have domains in the cart and no plan in the cart
2. We have a plan in the cart and no domains in the cart

This means that our cart contains a plan and domains, or it contains neither (but has other products).

Given that domains have a 4-day refund policy, and plans either have a 7-day or 14-day refund policy, my assumption is that we're intentionally vague for that case.

However, we're specifically aiming for the case where we have non-plan and non-domain products in the cart. Even for this case, I think we need to be careful, because the user may have other products with different refund windows in the cart. One example might be Google Workspace for example.com (which is annual) and Professional Email for example.net (which is monthly).

I think we need to change the logic to check for the refund windows of all products in the cart, and then display a specific number if the refund windows are all the same.

However, I don't see `bill_period` in the shopping cart response data represented in the `ResponseCartProduct` type definition. @sirbrillig, is that intentional? And is it OK for us to infer the refund window from all products based on their billing period? I believe this logic is specifically for WPCOM, so I assume this logic applies to WPCOM products.
```suggestion
	} else if ( ! hasPlanInCart && ! hasDomainsInCart ) {
		const cartRefundWindows = responseCarts.products.reduce( ( refundWindows, product ) =>
		{
			if ( PLAN_MONTHLY_PERIOD === product.bill_period ) {
				return {
					...refundWindows,
					7: true,
				};
			}

			if ( [ PLAN_ANNUAL_PERIOD, PLAN_BIANNUAL_PERIOD ].some( period => period === product.bill_period ) ) {
				return {
					...refundWindows,
					14: true,
				};
			}
			
			return refundWindows;
		},
		{}
		);

		// If we only have one refund window for all products, use that window
		if ( Object.keys( cartRefundWindows ).length === 1 ) {
			refundDays = Object.keys( cartRefundWindows ).pop();
		 }
		 // Otherwise keep the text generic
	}
```

Reply:
> However, I don't see bill_period in the shopping cart response data represented in the ResponseCartProduct type definition. @sirbrillig, is that intentional? 

Not that I know of. It's likely that we just haven't needed that property anywhere (in TS code) and so it never got added to the type? Feel free to add it.

> And is it OK for us to infer the refund window from all products based on their billing period? 

I think there's probably some nuance to that question based on what you're going to do with that data but in general, yes, it should be safe. That's what I'd use if I had a cart item. The other option is to use the massive client-side hard-coded data of product data that you can access with [`getPlan`](https://github.com/Automattic/wp-calypso/blob/1ac1cfc7036951ce324d54bfdf1fdc548d5a0dd2/packages/calypso-products/src/main.js#L41) and its peers (careful, there's also a Redux selector by that name).

Reply:
I've lost this conversation. @daledupreez The problem with this bug was that we were not showing the correct refund window (and support texts) when we were buying email, for a site that already had plan. In this case, the code won't fall into your conditions.